### PR TITLE
2.x: Cleaunp - rename fields to upstream and downstream

### PR DESCRIPTION
**Pull request by akarnokd**
This PR cleans up the field namings and some local variable namings:

- Use `upstream` for the connection `Disposable` or `Subscription` instead of `d`, `s`, etc., including `AtomicReference<*>` declarations.
- Use `downstream` for the consumer field name in operators instead of `actual` for example.
- Fix one-two letter abbreviated naming to match the type, such as `Disposable s` -> `Disposable d`, etc.

The `CheckLocalVariablesInTests` has been extended with the relevant regexp checks.

### CODE REVIEW

File: src/main/java/io/reactivex/MaybeEmitter.java [hunk 1]
```diff
@@ -68,14 +68,14 @@
 
     /**
      * Sets a Disposable on this emitter; any previous Disposable
-     * or Cancellation will be unsubscribed/cancelled.
-     * @param s the disposable, null is allowed
+     * or Cancellable will be disposed/cancelled.
```
Comment (artem-zinnatullin) [inline src/main/java/io/reactivex/MaybeEmitter.java]: `{@link`?

File: src/test/java/io/reactivex/validators/CheckLocalVariablesInTests.java [hunk 2]
```diff
@@ -26,11 +26,21 @@
  * <li>{@code TestObserver} named as {@code ts*}</li>
  * <li>{@code PublishProcessor} named as {@code ps*}</li>
  * <li>{@code PublishSubject} named as {@code pp*}</li>
+ * <li>{@code Subscription} with single letter name such as "s" or "d"</li>
```
Comment (artem-zinnatullin) [inline src/test/java/io/reactivex/validators/CheckLocalVariablesInTests.java:29]: 👍

**Review by artem-zinnatullin — APPROVED** [review]
Very welcomed change, thanks

**Comment by akarnokd** [comment]
Thanks @artem-zinnatullin for your diligent review of this PR.

**Review by vanniktech — COMMENTED** [review]
Very nice :+1:

**Comment by davidmoten** [comment]
:+1: very helpful thanks

File: src/main/java/io/reactivex/internal/observers/BasicFuseableObserver.java [hunk 3]
```diff
@@ -28,13 +28,13 @@
 public abstract class BasicFuseableObserver<T, R> implements Observer<T>, QueueDisposable<R> {
 
     /** The downstream subscriber. */
-    protected final Observer<? super R> actual;
```
Comment (codefromthecrypt) [inline src/main/java/io/reactivex/internal/observers/BasicFuseableObserver.java:31]: fyi these are protected fields so will break folks. We'll try and work around this, just saying. cc @kojilin

Comment (akarnokd) [reply src/main/java/io/reactivex/internal/observers/BasicFuseableObserver.java:31]: Internal.

https://github.com/ReactiveX/RxJava#ioreactivexinternal

**Comment by codefromthecrypt** [comment]
>
> In
> src/main/java/io/reactivex/internal/observers/BasicFuseableObserver.java
> <https://github.com/ReactiveX/RxJava/pull/6129#discussion_r225166066>:
>
> Internal.
>
> https://github.com/ReactiveX/RxJava#ioreactivexinternal
>
YIKES we have a lot of use of internal types, just didn't notice until you
mentioned. Will go through and rewrite a bunch of things. Thanks for the
pointer.

**Pull request merged**